### ISSUE TITLE
Resume training with deepspeed resets learning rate

### ISSUE THREAD

**Issue by BiEchi** [huggingface/transformers#27710]
### System Info

- `transformers` version: 4.32.0.dev0
- Platform: Linux-3.10.0-1160.31.1.el7.x86_64-x86_64-with-glibc2.17
- Python version: 3.9.0
- Huggingface_hub version: 0.16.4
- Safetensors version: 0.3.1
- Accelerate version: 0.21.0
- Accelerate config:    not found
- PyTorch version (GPU?): 2.0.1+cu117 (False)
- Tensorflow version (GPU?): not installed (NA)
- Flax version (CPU?/GPU?/TPU?): not installed (NA)
- Jax version: not installed
- JaxLib version: not installed
- Using GPU in script?: <fill in>
- Using distributed or parallel set-up in script?: <fill in>

### Who can help?

@pacman100 

### Information

- [X] The official example scripts
- [ ] My own modified scripts

### Tasks

- [ ] An officially supported task in the `examples` folder (such as GLUE/SQuAD, ...)
- [X] My own task or dataset (give details below)

### Reproduction

Launch script uses `run_mlm.py`:
```
python transformers/examples/pytorch/language-modeling/run_mlm.py \
        --resume_from_checkpoint $CKPT_TO_RESUME \
        --config_name $MODEL \
        --tokenizer_name $TOKENIZER \
        --dataset_name $PT_DATASET \
        --max_steps $MAX_STEPS \
        --preprocessing_num_workers 32 \
        --logging_steps $LOG_STEPS \
        --cache_dir $CACHE_DIR \
        --warmup_steps $WARMUP_STEPS \
        --learning_rate $PT_PEAK_LR \
        --lr_scheduler_type $PT_LR_DECAY \
        --output_dir $OUTPUT_DIR \
        --overwrite_output_dir \
        --deepspeed $DS_CONFIG
```
Deepspeed config:
```
{
    "zero_optimization": {
        "stage": 2,
        "allgather_partitions": true,
        "allgather_bucket_size": 2e8,
        "overlap_comm": true,
        "reduce_scatter": true,
        "reduce_bucket_size": 2e8,
        "contiguous_gradients": true
    },
    "fp16": {
        "enabled": false
    },
    "bf16": {
        "enabled": true
    },  
    "train_micro_batch_size_per_gpu": "auto",
    "gradient_accumulation_steps": "auto",
    "train_batch_size": "auto"
}
```
When resuming from deepspeed checkpoint, learning rate resets:
<img width="655" alt="image" src="https://github.com/huggingface/transformers/assets/60613238/6d9b3043-1b46-4505-bfeb-81eec0736aed">
If I comment out the deepspeed config (which means that Trainer gets rid of DS), the problem resolves:
<img width="654" alt="image" src="https://github.com/huggingface/transformers/assets/60613238/1ca4d189-f58c-4237-9246-6d6d0bfc8f6b">

Is there anything wrong with my deepspeed config?

### Expected behavior

The learning rate shoudn't reset.

**Comment by BiEchi** [comment]
Solved, we should add the specification for scheduler in the config json for deepspeed as well.
